tests/xharness/TestLabel.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.Diagnostics;
+using System.Linq;
+using Mono.Unix.Native;
+
+#nullable enable
+
+namespace Xharness {
+	[AttributeUsage (AttributeTargets.Field)]
+	public class LabelAttribute : Attribute {
+		public string Label { get; }
+
+		public LabelAttribute (string label)
+		{
+			Label = label;
+		}
+	}
+
+	[Flags]
+	public enum TestLabel {
+		[Label ("none")]
+		None = 0,
+		[Label ("bcl")]
+		Bcl = 1 << 0,
+		[Label ("mac")]
+		Mac = 1 << 1,
+		[Label ("ios")]
+		iOS = 1 << 2,
+		[Label ("ios-32")]
+		iOS32 = 1 << 3,
+		[Label ("ios-64")]
+		iOS64 = 1 << 4,
+		[Label ("ios-extensions")]
+		iOSExtension = 1 << 5,
+		[Label ("ios-simulator")]
+		iOSSimulator = 1 << 6,
+		[Label ("old-simulator")]
+		OldiOSSimulator = 1 << 7,
+		[Label ("device")]
+		Device = 1 << 8,
+		[Label ("xtro")]
+		Xtro = 1 << 9,
+		[Label ("cecil")]
+		Cecil = 1 << 10,
+		[Label ("docs")]
+		Docs = 1 << 11,
+		[Label ("bcl-xunit")]
+		BclXUnit = 1 << 12,
+		[Label ("bcl-nunit")]
+		BclNUnit = 1 << 13,
+		[Label ("mscorlib")]
+		Mscorlib = 1 << 14,
+		[Label ("non-monotouch")]
+		NonMonotouch = 1 << 15,
+		[Label ("monotouch")]
+		Monotouch = 1 << 16,
+		[Label ("dotnet")]
+		Dotnet = 1 << 17,
+		[Label ("maccatalyst")]
+		MacCatalyst = 1 << 18,
+		[Label ("tvos")]
+		tvOS = 1 << 19,
+		[Label ("watchos")]
+		watchOS = 1 << 20,
+		[Label ("mmp")]
+		Mmp = 1 << 21,
+		[Label ("msbuild")]
+		Msbuild = 1 << 22,
+		[Label ("mtouch")]
+		Mtouch = 1 << 23,
+		[Label ("btouch")]
+		Btouch = 1 << 24,
+		[Label ("mac-binding-project")]
+		MacBindingProject = 1 << 25,
+		[Label ("system-permission")]
+		SystemPermission = 1 << 26,
+		[Label ("all")]
+		All = ~None,
```

Comment:
Minor, but you probably don't want to do `All = ~None` because it really messes up debugging since `ToString` won't be able to do as much work for you as you'd like and it also violates the principal of least astonishment - if I OR each individual flag together and compare to `All` it won't match. Better to OR the values you want.